tests/integration_tests/test_upgrade.py

```diff
@@ -1,182 +1,182 @@
-    contract, _ = deploy_contract(w3, CONTRACTS["TestERC20A"])
-    old_height = w3.eth.block_number
-    old_balance = w3.eth.get_balance(ADDRS["validator"], block_identifier=old_height)
-    old_base_fee = w3.eth.get_block(old_height).baseFeePerGas
-    old_erc20_balance = contract.caller.balanceOf(ADDRS["validator"])
-    print("old values", old_height, old_balance, old_base_fee)
-
-    target_height = w3.eth.block_number + 10
-    print("upgrade height", target_height)
-
-    plan_name = "integration-test-upgrade"
-    rsp = cli.gov_propose(
-        "community",
-        "software-upgrade",
-        {
-            "name": plan_name,
-            "title": "upgrade test",
-            "description": "ditto",
-            "upgrade-height": target_height,
-            "deposit": "10000aphoton",
-        },
-    )
-    assert rsp["code"] == 0, rsp["raw_log"]
-
-    # get proposal_id
-    ev = parse_events(rsp["logs"])["submit_proposal"]
-    proposal_id = ev["proposal_id"]
-
-    rsp = cli.gov_vote("validator", proposal_id, "yes")
-    assert rsp["code"] == 0, rsp["raw_log"]
-    # rsp = custom_ethermint.cosmos_cli(1).gov_vote("validator", proposal_id, "yes")
-    # assert rsp["code"] == 0, rsp["raw_log"]
-
-    proposal = cli.query_proposal(proposal_id)
-    wait_for_block_time(cli, isoparse(proposal["voting_end_time"]))
-    proposal = cli.query_proposal(proposal_id)
-    assert proposal["status"] == "PROPOSAL_STATUS_PASSED", proposal
-
-    # update cli chain binary
-    custom_ethermint.chain_binary = (
-        Path(custom_ethermint.chain_binary).parent.parent.parent
-        / f"{plan_name}/bin/ethermintd"
-    )
-    cli = custom_ethermint.cosmos_cli()
-
-    # block should pass the target height
-    wait_for_block(cli, target_height + 1, timeout=480)
-    wait_for_port(ports.rpc_port(custom_ethermint.base_port(0)))
-
-    # test migrate keystore
-    cli.migrate_keystore()
-
-    # check basic tx works after upgrade
-    wait_for_port(ports.evmrpc_port(custom_ethermint.base_port(0)))
-
-    receipt = send_transaction(
-        w3,
-        {
-            "to": ADDRS["community"],
-            "value": 1000,
-            "maxFeePerGas": 1000000000000,
-            "maxPriorityFeePerGas": 10000,
-        },
-    )
-    assert receipt.status == 1
-
-    # check json-rpc query on older blocks works
-    assert old_balance == w3.eth.get_balance(
-        ADDRS["validator"], block_identifier=old_height
-    )
-    assert old_base_fee == w3.eth.get_block(old_height).baseFeePerGas
-
-    # check eth_call on older blocks works
-    assert old_erc20_balance == contract.caller(
-        block_identifier=target_height - 2
-    ).balanceOf(ADDRS["validator"])
-    p = json.loads(
-        cli.raw(
-            "query",
-            "ibc",
-            "client",
-            "params",
-            home=cli.data_dir,
-        )
-    )
-    assert p == {"allowed_clients": ["06-solomachine", "07-tendermint", "09-localhost"]}
+# import configparser
```

Comment:
Why not just completely remove this file?

Reply:
I kept it just for historical reasons, so when we work on https://github.com/zeta-chain/node/issues/2569 we can consider it to be part of e2e tests. We can delete it without problem.